Source/CarthageKit/Xcode.swift

```diff
@@ -789,14 +789,19 @@ private func mergeBuildProductsIntoDirectory(firstProductSettings: BuildSettings
 		}
 }
 
+
+/// A callback function used to determine whether or not an SDK should be built
+public typealias SDKFilterCallback = (sdks: [SDK], scheme: String, configuration: String, project: ProjectLocator) -> [SDK];
+
 /// Builds one scheme of the given project, for all supported SDKs.
 ///
 /// Returns a signal of all standard output from `xcodebuild`, and a signal
 /// which will send the URL to each product successfully built.
-public func buildScheme(scheme: String, withConfiguration configuration: String, inProject project: ProjectLocator, #workingDirectoryURL: NSURL) -> SignalProducer<TaskEvent<NSURL>, CarthageError> {
+public func buildScheme(scheme: String, withConfiguration configuration: String, inProject project: ProjectLocator, canBuildSDK: SDKFilterCallback? = nil, #workingDirectoryURL: NSURL, sendWarning: (String -> Void)? = nil) -> SignalProducer<TaskEvent<NSURL>, CarthageError> {
```

Comment:
I had started out that way, but then the default gets repeated in the signature of all the methods in the call chain, whereas this way allows me to specify the default behavior one time, closest to where it's used. Do you know of a way around repeating the ` = { $0 }` in each of the other methods (or is that an acceptable tradeoff to you)?

As a side note, ` = { $0 }` doesn't compile, though this does: `= { (sdks, _, _, _) in sdks }`. I've [asked about that](http://stackoverflow.com/q/32216909/105717) on Stack Overflow, as it makes no sense to me.